userspace/libsinsp/test/sinsp_utils.ut.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright (C) 2023 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <gtest/gtest.h>
+#include "utils.h"
+
+// copy_and_sanitize_path is not exported in utils.h
+void copy_and_sanitize_path(char* target, char* targetbase, const char* path, char separator);
+
+TEST(sinsp_utils_test, copy_and_sanitize_path)
+{
+	char target[SCAP_MAX_PATH_SIZE];
+
+	std::string path = "/dir/term";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ(path, std::string(target));
+
+	path = "/dir/term/";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/dir/term", std::string(target));
+
+	path = "/dir/../term";
+	copy_and_sanitize_path(target, target, path.c_str(), '/');
+	EXPECT_EQ("/term", std::string(target));
```

Comment:
Unfortunately this is wrong (but I'm not sure we can do better in the presence of mount namespaces):

```
root@hex:/tmp# ln -s /etc xxx
root@hex:/tmp# ls -lda /tmp/xxx/../boot
drwxr-xr-x 3 root root 4096 mar  9 06:10 /tmp/xxx/../boot
root@hex:/tmp# LANG=C ls -lda /tmp/xxx/boot
ls: cannot access '/tmp/xxx/boot': No such file or directory
```

`readlink -f` gets this right but at the cost of accessing the filesystem (it's no longer a simple string operation):

```
root@hex:/tmp# readlink -f /tmp/xxx/../boot
/boot
```

(looks like we've had this bug forever, so I'm just pointing this out)

Reply:
Good point, @gnosek! @LucaGuerra @Andreagit97 Any suggestions here?

Reply:
Yeah, I don't think it's solvable here :thinking: . It's been a part of the behavior of libsinsp since the beginning but it can indeed create incorrect paths. And of course, we don't want to call `readlink` here.

Reply:
@araujof definitely out of scope for this :)